mcs/class/System/System.CodeDom.Compiler/CompilerCollection.cs

```diff
@@ -50,32 +50,17 @@ static CompilerCollection ()
 			properties = new ConfigurationPropertyCollection ();
 			compiler_infos = new List <CompilerInfo> ();
 			compiler_languages = new Dictionary <string, CompilerInfo> (16, StringComparer.OrdinalIgnoreCase);
-			compiler_extensions = new Dictionary <string, CompilerInfo> (6, StringComparer.OrdinalIgnoreCase);
+			compiler_extensions = new Dictionary <string, CompilerInfo> (4, StringComparer.OrdinalIgnoreCase);
```

Comment:
nit: this is for all extensions so the Dictionary capacity would need to be >= 7.

Reply:
The extensions are `new [] { ".cs" }` so there are 2 right now

Reply:
Then the way we're using CompilerInfo is wrong since extensions are the last ctor argument: https://github.com/mono/mono/blob/4e6f5f10fd26a50981b7e928babf910082d09192/mcs/class/referencesource/System/compmod/system/codedom/compiler/CompilerInfo.cs#L108.